doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,950 @@
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+st.write("# Calculation Runner")
+
+option = st.radio("Which calculation would you like to perform?", ("A", "B"))
+
+st.write("You chose: ", option)
+if option == "A":
+    if st.button("Press to run calculation"):
+        with st.spinner("Running... Please wait!"):
+            time_start = time.perf_counter()
+            result = calculation_a()
+            time_end = time.perf_counter()
+            st.write(f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete""")
+
+    else:
+        st.write("Calculation A did not run yet")
+
+elif option == "B":
+    if st.button("Press to run calculation"):
+        with st.spinner("Running... Please wait!"):
+            time_start = time.perf_counter()
+            result = calculation_b()
+            time_end = time.perf_counter()
+            st.write(f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete""")
+    else:
+        st.write("Calculation B did not run yet")
+```
+
+The app looks like
+
+![Streamlit multiple updates example](https://user-images.githubusercontent.com/42288570/243399616-4f172e4b-ace9-4761-b19a-757a532bafbc.gif)
+
+#### Panel Multiple Updates Example
+
+With Panel you will use a *generator function* to update a component multiple times during code execution.
+
+```python
+import panel as pn
+import time
+import random
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+pn.state.template.param.update(site="Panel", title="Calculation Runner")
+
+def calculation_message(calculation):
+    return f"You chose: {calculation}"
+
+def calculation_a():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+def calculation_result(running, calculation):
+    if not running:
+        yield "Calculation did not run yet"
+        return # This will break the execution
+
+    yield pn.Row(pn.indicators.LoadingSpinner(value=True, width=50, height=50, align="center"), "Running... Please wait!", align="center")
+
+    if calculation=="A":
+        func = calculation_a
+    else:
+        func = calculation_b
+
+    time_start = time.perf_counter()
+    result = func()
+    time_end = time.perf_counter()
+    yield f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete"""
+
+
+calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
+run_input = pn.widgets.Button(name="Press to run calculation", icon="caret-right", button_type="primary", sizing_mode="fixed", width=250)
+
+pn.Column(
+    "Which calculation would you like to perform?",
+    calculation_input,
+    pn.bind(calculation_message, calculation_input),
+    run_input,
+    pn.bind(calculation_result, run_input, calculation_input)
```

Comment:
Renamed functions to verbs and some dedent cleanup

```suggestion
import time
import random
from textwrap import dedent

import panel as pn

pn.extension(sizing_mode="stretch_width", template="bootstrap")
pn.state.template.param.update(site="Panel", title="Calculation Runner")


def notify_choice(calculation):
    return f"You chose: {calculation}"


def calculation_a():
    time.sleep(1.5)
    return random.randint(0, 100)


def calculation_b():
    time.sleep(1.5)
    return random.randint(0, 100)


def run_calculation(running, calculation):
    if not running:
        yield "Calculation did not run yet"
        return  # This will break the execution
    yield pn.Row(
        pn.indicators.LoadingSpinner(value=True, width=50, height=50, align="center"),
        "Running... Please wait!",
        align="center",
    )
    if calculation == "A":
        func = calculation_a
    else:
        func = calculation_b
    time_start = time.perf_counter()
    result = func()
    time_end = time.perf_counter()
    yield dedent(
        f"""
        Done!
        Result: {result}
        The function took {time_end - time_start:1.1f} seconds to complete
        """
    )


calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
run_input = pn.widgets.Button(
    name="Press to run calculation",
    icon="caret-right",
    button_type="primary",
    sizing_mode="fixed",
    width=250,
)
pn.Column(
    "Which calculation would you like to perform?",
    calculation_input,
    pn.bind(notify_choice, calculation_input),
    run_input,
    pn.bind(run_calculation, run_input, calculation_input),
).servable()
```